src/reader/decoder.rs

```diff
@@ -38,16 +72,15 @@ impl fmt::Display for DecodingError {
 impl error::Error for DecodingError {
     fn description(&self) -> &str {
         match *self {
-            DecodingError::Format(ref d) => d,
-            DecodingError::Internal(ref d) => d,
+            DecodingError::Format(ref err) => err.description(),
             DecodingError::Io(ref err) => err.description(),
         }
     }
```

Comment:
I wasnt too sure whether those should be kept or not due to msrv. But you are right there isn't really a point in keeping those. I also just realized I implemented `cause` instead of `source` in DecodingError, fixed that as well.